sessions/01-exploration-tools/README.md

```diff
@@ -0,0 +1,714 @@
+
+Let's illustrate how strings can be useful in a simple context. Try out crackme1 from the tutorial archive (01-tutorial-strings directory):
+
+```c
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+ 
+int my_strcmp(char *s1, char *s2)
+{
+	size_t i, len = strlen(s1);
+    if (len == 0)
+	return -1;
+    for (i = 0; i < len; i++)
+        if (s1[i] != s2[i])
+	    return -1;
+    return 0;
+}
+
+int main()
+{
+    char buf[1000];
+ 
+    printf("Password:\n");
+    if (fgets(buf, 1000, stdin) == NULL)
+	exit(-1);
+
+    buf[strlen(buf) - 1] = '\0';
+
+    if (!my_strcmp(buf, ???????????????????????)) {
+        printf("Correct!\n");
+    } else
+	printf("Nope!\n");
+
+    return 0;
+}
+```
+
+The password has been redacted from the listing but you can retrieve it with `strings`. Try it out!
+
+>If you need to retrieve the offset of a string in a binary file, you may use the `-t` option of `strings`. For example, to print out the offset of the `Correct` string (in hexadecimal), you would issue the command
+`strings -t x crackme1 | grep Correct`
+
+### 2. Tutorial - Execution tracing (ltrace and strace)
+
+[ltrace](https://man7.org/linux/man-pages/man1/ltrace.1.html) is an utility that can list the calls made to library functions made by a program, or the [syscalls](https://man7.org/linux/man-pages/man2/syscalls.2.html) a program makes. A syscall is a function that uses services exposed by the kernel, not by some separate library.
+
+The way strace works is with the aid of a special syscall, called [ptrace](https://man7.org/linux/man-pages/man2/ptrace.2.html). This single syscall forms the basis for most of the functionality provided by ltrace, strace, gdb and similar tools that debug programs. It can receive up to 4 arguments: the operation, the PID to act on, the address to read/write and the data to write. The functionality exposed by ptrace() is massive, but think of any functionality you've seen in a debugger:
+
+* tach/detach to a process
+* set breakpoints
+* continue a stopped program
+* read/write registers
+* act on signals
+* register syscalls
+
+A tool like `strace` only traces syscalls and reads registers in order to provide some pretty printing strictly concerning the syscalls of the traced process. However, ltrace provides further functionality and gathers information about all library calls. Here's how `ltrace` does its magic:
+
+* it reads the tracee memory and parses it in order to find out about loaded symbols
+* it makes a copy of the binary code pertaining to a symbol using a PTRACE_PEEKTEXT directive of ptrace()
+* it injects a breakpoint using a PTRACE_POKETEXT directive of ptrace()
+* it listens for a SIGTRAP which will be generated when the breakpoint is hit
+* when the breakpoint is hit, ltrace can examine the stack of the tracee and print information such as function name, parameters, return codes, etc.
```

Comment:
Add inline code blocks to the flags and function calls. It makes them a lot more readable.